test/widgets/message_list_test.dart

```diff
@@ -648,6 +645,31 @@ void main() {
       check(isMarkAsReadButtonVisible(tester)).isFalse();
     });
 
+    testWidgets('messages don\'t shift position', (WidgetTester tester) async {
+      final message = eg.streamMessage(flags: []);
+      final unreadMsgs = eg.unreadMsgs(streams:[
+        UnreadStreamSnapshot(topic: message.subject, streamId: message.streamId,
+          unreadMessageIds: [message.id])
+      ]);
+      await setupMessageListPage(tester,
+        messages: [message], unreadMsgs: unreadMsgs);
+      check(isMarkAsReadButtonVisible(tester)).isTrue();
+      check(tester.widgetList(find.byType(MessageItem)).length).equals(1);
+      final before = tester.getTopLeft(find.byType(MessageItem)).dy;
+
+      store.handleEvent(UpdateMessageFlagsAddEvent(
+        id: 1,
+        flag: MessageFlag.read,
+        messages: [message.id],
+        all: false,
+      ));
+      await tester.pumpAndSettle();
+      check(isMarkAsReadButtonVisible(tester)).isFalse();
+      check(tester.widgetList(find.byType(MessageItem)).length).equals(1);
+      final after = tester.getTopLeft(find.byType(MessageItem)).dy;
+      check(before).equals(after);
```

Comment:
Check actual equals expected:
```suggestion
      check(after).equals(before);
```

That way `check` is able to give accurate output if the check fails. (To see the failure, use `git checkout @~ lib/` and then `flutter test test/widgets/message_list_test.dart`, to run your new test on the old code.)

Reply:
Thanks for the explanation. Fixed.